libsolidity/parsing/Parser.cpp

```diff
@@ -599,13 +606,16 @@ ASTPointer<ASTNode> Parser::parseFunctionDefinition(bool _freeFunction)
 			m_scanner->next();
 		}
 		else
-			name = expectIdentifierToken();
+		{
+			tie(name, nameLocation) = expectIdentifierWithLocation();
+		}
 	}
 	else
 	{
 		solAssert(kind == Token::Constructor || kind == Token::Fallback || kind == Token::Receive, "");
 		m_scanner->next();
 		name = make_shared<ASTString>();
+		nameLocation = currentLocation();
```

Comment:
is this empty?

Reply:
yes, just like name is empty. I dropped both lines to better designate the "empty"-state.